andela/travel_tool_back
# #162008362 Fix checklist submission issues

#### What does this PR do?
This PR restructures checklist submission response data and rename checklist submissions endpoint to meet standard.

#### Description of Task to be completed?
- Restructure checklist submission response data
- Refactor get/post checklist submission endpoint
- Refactor get/post checklist submission method
- Refactor calculate checklist percentage method
- Refactor tests to accommodate changes
- Add migration files for default checklist items and resources.

#### How should this be manually tested?
- Clone the repo - `git clone https://github.com/andela/travel_tool_back.git`
- Enter into the project directory - `cd travel_tool_back`
- switch to this branch - `bg-fix-checklist-submissions-162008362`
- Rollback, run database migrations and seed database - `yarn db:rollmigrate`
- Install dependencies and start the server - `make start`
-  [Start](https://github.com/andela/travel_tool_front/pull/321)  the frontend.
- Navigate to [travela](http://travela-local.andela.com:3000/#) in your browser and login
- Navigate to the `UserRoles` table and add yourself as a `manager`
- Navigate to the `travela-app` in your browser and refresh
- Make a `New Request` and choose yourself as the `manager`
- Approve the request
- Open `Postman`
- Grab your token from the application tab in Developers tab
- On postman add the token in headers
- Navigate to the `database`, on the checklistItem table copy the `id` of a `checklistItem and its `tripId`
- Navigate to the application and copy the `approved request Id`
- - On `postman` Send a `POST` request to `http://localhost:5000/api/v1/checklists/:requestId/submissions/:checklistItemId`
- You should receive a successful response
- Also send a `GET` request to `http://localhost:5000/api/v1/checklists/:requestId/submissions`
- You should get all checklist Item associated with the `requestId`

#### Any background context you want to provide?
- the POST endpoint payload should be structured like this- 
`
{
	"tripId": "",
	"file": ""
}
`

#### What are the relevant pivotal tracker stories?
[#162008362](https://www.pivotaltracker.com/story/show/162008362)

#### Screenshots (if appropriate)
<img width="1440" alt="screenshot 2018-11-21 at 11 04 46 am" src="https://user-images.githubusercontent.com/27569442/48834389-fd3b0a80-ed7d-11e8-9901-c7fa1e47a306.png">

<img width="1440" alt="screenshot 2018-11-21 at 11 09 07 am" src="https://user-images.githubusercontent.com/27569442/48834373-f4e2cf80-ed7d-11e8-9241-e7d55adda578.png">


#### Questions:
N/A
